sources/libs/fs/ext2/ext2.c

```diff
@@ -43,120 +100,417 @@ static size_t get_entry_in_indirect_block(size_t entry_per_block, int depth)
     return res;
 }
 
-static uint64_t get_inode_size(Ext2SuperBlock const *super_block, Ext2FsInode const *inode)
+static uint64_t inode_size(Ext2Fs const *self, Ext2FsInode const *inode)
 {
     uint64_t len = inode->block.size_low;
-    if (super_block->read_only_feature & EXT2_FEAT_64BIT_FILE_SIZE)
+    if (self->super_block->read_only_feature & EXT2_FEAT_64BIT_FILE_SIZE)
     {
         len |= (uint64_t)inode->block.size_upper << 32;
     }
 
     return len;
 }
 
+FsResult ext2_inode_rw(Ext2Fs *self, Ext2FsInode *inode, Ext2FsInodeId id, bool write)
```

Comment:
I don't like function behaving completely differently based on a boolean, most of the time this mean the fonctions need to be splitted

Reply:
It's to avoid repetitive code, and this function is splitted in 2: ext2_write_inode & ext2_inode

Reply:
Well then just isolate the common code in another function

Reply:
I don't see the point of doing it if in the end we only use it in 2 functions.  (it should be static tho)